nodes/stampzilla-linux/volume.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	volume "github.com/itchyny/volume-go"
+	"github.com/sirupsen/logrus"
+	"github.com/stampzilla/stampzilla-go/nodes/stampzilla-server/models/devices"
+)
+
+func monitorVolume() {
+	dev := &devices.Device{
+		Name:   "Audio",
+		ID:     devices.ID{ID: "audio"},
+		Online: true,
+		Traits: []string{"OnOff", "Volume"},
+		State: devices.State{
+			"on":     false,
+			"volume": 0,
+		},
+	}
+
+	for {
+		vol, err := volume.GetVolume()
+		if err != nil {
+			logrus.Errorf("get volume failed: %+v", err)
+			return
+		}
+
+		mute, err := volume.GetMuted()
+		if err != nil {
+			logrus.Errorf("get mute failed: %+v", err)
+			return
+		}
+
+		dev.State["on"] = !mute
+		dev.State["volume"] = float64(vol) / 100
+		n.AddOrUpdate(dev)
```

Comment:
never use `AddOrUpdate` if you know the device exists. Dont modify dev.State it does not lock. 

Always create new state and then UpdateState

like this: 
```
newState := make(devices.State)
newState["volume"] = float64(vol) / 100
dev.State["on"] = !mute
n.UpdateState(dev.ID.ID, newState)

```

It will look properly and only update the states and send it to server if there was a diff.